File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.go
func minimumAbsDifference(arr []int) [][]int {
	sort.Slice(arr, func(i, j int) bool { return arr[i] < arr[j] })

	minDiff := (1 << 31) - 1
	for i := 1; i < len(arr); i++ {
		minDiff = min(minDiff, arr[i]-arr[i-1])
	}

	ans := [][]int{}
	for i := 1; i < len(arr); i++ {
		diff := arr[i] - arr[i-1]
		if diff == minDiff {
			ans = append(ans, []int{arr[i-1], arr[i]})
		}
	}

	return ans
}

// array, sorting
// time: O(nlog(n))
// space: O(n)

File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()

        min_diff = sys.maxsize
        for i in range(1, len(arr)):
            min_diff = min(min_diff, arr[i] - arr[i - 1])
        
        ans = []
        for i in range(1, len(arr)):
            diff = arr[i] - arr[i - 1]
            if diff == min_diff:
                ans.append([arr[i - 1], arr[i]])
        
        return ans

# array, sorting
# time: O(nlog(n))
# space: O(n)